futures-util/src/try_stream/mod.rs

```diff
@@ -572,8 +576,42 @@ pub trait TryStreamExt: TryStream {
     /// ```
     #[cfg(feature = "compat")]
     fn compat(self) -> Compat<Self>
-        where Self: Sized + Unpin,
+    where
+        Self: Sized + Unpin,
     {
         Compat::new(self)
     }
+
+    /// Adapter that converts this stream into an [`AsyncRead`].
+    ///
+    /// Note that because `into_async_read` moves the stream, the [`Stream`] type must be
+    /// [`Unpin`]. If you want to use `into_async_read` with a [`!Unpin`](Unpin) stream, you'll
+    /// first have to pin the stream. This can be done by boxing the stream using [`Box::pinned`]
+    /// or pinning it to the stack using the `pin_mut!` macro from the `pin_utils` crate.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use futures::executor::block_on;
+    /// use futures::future::lazy;
+    /// use futures::stream::{self, StreamExt, TryStreamExt};
+    /// use futures_io::AsyncRead;
+    /// use std::io::Error;
+    ///
+    /// let stream = stream::iter(vec![Ok::<Vec<u8>, Error>(vec![1, 2, 3, 4, 5])]);
+    /// let mut reader = stream.into_async_read();
+    /// let mut buf = vec![0; 3];
+    ///
+    /// block_on(lazy(|ctx| {
```

Comment:
Consider maybe using something like `read_to_end` instead, and show it w/ `await!`? typically we try to give examples in terms of the `async`/`await!` usage

Reply:
(also an `assert!` and the end showing that the operation completed successfully would be welcome ;) )